commit 6091b58b60f532d8e3f317b99f7ea85d99853db3 (HEAD, master)
Author: Bill Nottingham <notting@redhat.com>
Date:   Wed Oct 31 13:31:48 2012 -0400

    Drop fedora-storage-init; conflict with older lvm & dmraid so we get the required versions that allow us to do so. (<prajnoha@redhat.com>)

diff --git a/Makefile b/Makefile
index e8f3fb5..233ab00 100644
--- a/Makefile
+++ b/Makefile
@@ -96,8 +96,6 @@ install:
 	ln -s ../fedora-autorelabel-mark.service $(ROOT)/lib/systemd/system/basic.target.wants
 	ln -s ../fedora-readonly.service $(ROOT)/lib/systemd/system/local-fs.target.wants
 	ln -s ../fedora-import-state.service $(ROOT)/lib/systemd/system/local-fs.target.wants
-	ln -s ../fedora-storage-init.service $(ROOT)/lib/systemd/system/local-fs.target.wants
-	ln -s ../fedora-storage-init-late.service $(ROOT)/lib/systemd/system/local-fs.target.wants
 
 	mkdir -p $(ROOT)/lib/tmpfiles.d
 	install -m 644 initscripts.tmpfiles.d $(ROOT)/lib/tmpfiles.d/initscripts.conf
diff --git a/initscripts.spec b/initscripts.spec
index 675debc..be577c9 100644
--- a/initscripts.spec
+++ b/initscripts.spec
@@ -18,6 +18,8 @@ Requires: bash >= 3.0
 Requires: sysvinit-tools >= 2.87-5
 Conflicts: systemd < 23-1
 Conflicts: systemd-units < 23-1
+Conflicts: lvm2 < 2.02.98-3
+Conflicts: dmraid < 1.0.0.rc16-18
 Requires: systemd
 Requires: iproute, /sbin/arping, findutils
 # Not strictly required, but nothing else requires it
diff --git a/systemd/fedora-storage-init b/systemd/fedora-storage-init
deleted file mode 100755
index 565d811..0000000
--- a/systemd/fedora-storage-init
+++ /dev/null
@@ -1,41 +0,0 @@
-#!/bin/bash
-#
-# Storage initialization
-
-. /etc/init.d/functions
-
-[ -z "${cmdline}" ] && cmdline=$(cat /proc/cmdline)
-
-if ! strstr "$cmdline" nompath && [ -f /etc/multipath.conf ] && \
-		[ -x /sbin/multipath ]; then
-	modprobe dm-multipath > /dev/null 2>&1
-	/sbin/multipath -u -v 0
-	if [ -x /sbin/kpartx ]; then
-		/sbin/dmsetup ls --target multipath --exec "/sbin/kpartx -u -a -p p" >/dev/null
-	fi
-fi
-
-if ! strstr "$cmdline" nodmraid && [ -x /sbin/dmraid ]; then
-	modprobe dm-mirror >/dev/null 2>&1
-	dmraidsets=$(LC_ALL=C /sbin/dmraid -s -c -i)
-	if [ "$?" = "0" ]; then
-		SAVEIFS=$IFS
-		IFS=$(echo -en "\n\b")
-		for dmname in $dmraidsets; do
-			if [[ "$dmname" == isw_* ]] && \
-			   ! strstr "$cmdline" noiswmd; then
-				continue
-			fi
-			/sbin/dmraid -ay -i --rm_partitions -p "$dmname" >/dev/null 2>&1
-			/sbin/kpartx -u -a -p p "/dev/mapper/$dmname"
-		done
-		IFS=$SAVEIFS
-	fi
-fi
-
-# Start any MD RAID arrays that haven't been started yet
-[ -r /proc/mdstat ] && [ -r /dev/md/md-device-map ] && /sbin/mdadm -IRs
-
-if [ -x /sbin/lvm ]; then
-	action $"Setting up Logical Volume Management:" /sbin/lvm vgchange -a y --sysinit
-fi
diff --git a/systemd/system/fedora-storage-init-late.service b/systemd/system/fedora-storage-init-late.service
deleted file mode 100644
index 69ad089..0000000
--- a/systemd/system/fedora-storage-init-late.service
+++ /dev/null
@@ -1,16 +0,0 @@
-[Unit]
-Description=Initialize storage subsystems (RAID, LVM, etc.)
-DefaultDependencies=no
-Conflicts=shutdown.target
-After=cryptsetup.target fedora-storage-init.service
-Before=local-fs.target shutdown.target
-Wants=fedora-wait-storage.service
-ConditionFileIsExecutable=|/sbin/lvm
-ConditionFileIsExecutable=|/sbin/dmraid
-ConditionPathExists=|/etc/multipath.conf
-
-[Service]
-ExecStart=/lib/systemd/fedora-storage-init
-Type=oneshot
-TimeoutSec=0
-RemainAfterExit=yes
diff --git a/systemd/system/fedora-storage-init.service b/systemd/system/fedora-storage-init.service
deleted file mode 100644
index 6d8e39f..0000000
--- a/systemd/system/fedora-storage-init.service
+++ /dev/null
@@ -1,16 +0,0 @@
-[Unit]
-Description=Initialize storage subsystems (RAID, LVM, etc.)
-DefaultDependencies=no
-Conflicts=shutdown.target
-After=fedora-wait-storage.service
-Before=local-fs.target shutdown.target
-Wants=fedora-wait-storage.service
-ConditionFileIsExecutable=|/sbin/lvm
-ConditionFileIsExecutable=|/sbin/dmraid
-ConditionPathExists=|/etc/multipath.conf
-
-[Service]
-ExecStart=/lib/systemd/fedora-storage-init
-Type=oneshot
-TimeoutSec=0
-RemainAfterExit=yes
diff --git a/systemd/system/fedora-wait-storage.service b/systemd/system/fedora-wait-storage.service
deleted file mode 100644
index 9d3b6ae..0000000
--- a/systemd/system/fedora-wait-storage.service
+++ /dev/null
@@ -1,18 +0,0 @@
-[Unit]
-Description=Wait for storage scan
-DefaultDependencies=no
-Conflicts=shutdown.target
-After=systemd-udev-settle.service
-Before=local-fs.target shutdown.target
-Wants=systemd-udev-settle.service
-
-[Service]
-ExecStart=-/sbin/rmmod scsi_wait_scan
-ExecStart=-/sbin/modprobe scsi_wait_scan
-ExecStart=-/sbin/rmmod scsi_wait_scan
-Type=oneshot
-TimeoutSec=0
-RemainAfterExit=yes
-StandardInput=null
-StandardOutput=null
-StandardError=null
